libraries/AP_Scripting/applets/RockBlock.lua

```diff
@@ -0,0 +1,752 @@
+            wind_xy:x(wind:x())
+            wind_xy:y(wind:y())
+            hl2.windspeed = math.abs(math.floor(wind_xy:length() * 5))
+            hl2.wind_heading = math.floor(wrap_360(wind_xy:angle()) / 2)
+        end
+        hl2.custom_mode = vehicle:get_mode()
+
+        if vehicle:get_wp_distance_m() ~= nil then
+            hl2.target_distance = math.floor(vehicle:get_wp_distance_m() / 10)
+        end
+        if mission:get_current_nav_index() ~= nil then
+            hl2.wp_num = mission:get_current_nav_index()
+        end
+        if vehicle:get_wp_bearing_deg() ~= nil then
+            hl2.target_heading = math.floor(wrap_360(
+                                                vehicle:get_wp_bearing_deg()) /
+                                                2)
+        end
+
+        -- failure flags
+        hl2.failure_flags = 0
+        if not ahrs:healthy() then
+            hl2.failure_flags = hl2.failure_flags + 4096 -- HL_FAILURE_FLAG_ESTIMATOR
+        end
+        if battery:num_instances() > 0 and not battery:healthy(0) then
+            hl2.failure_flags = hl2.failure_flags + 128 -- HL_FAILURE_FLAG_BATTERY
+        end
+        if gps:num_sensors() > 0 and gps:status(0) <= gps.NO_FIX then
+            hl2.failure_flags = hl2.failure_flags + 1 -- HL_FAILURE_FLAG_GPS
+        end
+        if (FWVersion:type() == 2 or FWVersion:type() == 3) and terrain:status() ==
+            terrain.TerrainStatusUnhealthy then
+            -- only for copter and plane
+            hl2.failure_flags = hl2.failure_flags + 64 -- HL_FAILURE_FLAG_TERRAIN
+        end
+        if not rc:has_valid_input() then
+            hl2.failure_flags = hl2.failure_flags + 256 -- HL_FAILURE_FLAG_RC_RECEIVER
+        end
+
+        hl2.heading = math.floor(wrap_360(math.deg(ahrs:get_yaw())) / 2)
+        hl2.throttle = math.floor(gcs:get_hud_throttle())
+        if ahrs:airspeed_estimate() ~= nil then
+            hl2.airspeed = math.abs(math.floor(ahrs:airspeed_estimate() * 5))
+        end
+        -- hl2.airspeed_sp = 0
+        hl2.groundspeed = math.abs(math.floor(
+                                       ahrs:groundspeed_vector():length() * 5))
+
+        hl2.temperature_air = math.floor(baro:get_external_temperature())
+        hl2.battery = battery:capacity_remaining_pct(0)
+
+        -- just sending armed state here for simplicity. Flight mode is in the custom_mode field
+        if arming:is_armed() then
+            hl2.custom0 = 129 -- MAV_MODE_FLAG_SAFETY_ARMED + MAV_MODE_FLAG_CUSTOM_MODE_ENABLED
+        else
+            hl2.custom0 = 1 -- MAV_MODE_FLAG_CUSTOM_MODE_ENABLED
+        end
+
+        local newpkt = mavlink.createMAVLink(hl2, mavlink.HIGH_LATENCY2)
+        if #newpkt > 50 then
+            gcs:send_text(3, "Rockblock: Tx packet > 50 bytes: " .. tostring(#newpkt))
+        end
+
+        -- send packet
+        rockblock.loadtxbuffer(newpkt)
+        
+    end
+
+    return HLSatcom, 100
+end
+
+return HLSatcom, 100
```

Comment:
add a pcall wrapper so it restarts on an exception, see other applets for example